src/varint.rs

```diff
@@ -0,0 +1,89 @@
+//! This module has been adapted from the [integer_encoding] crate. The
```

Comment:
Did you try asking them to entirely remove the `Send` bound? I don't think it is needed

Reply:
Ahh, yah, I think the `varint` crate enforces a `Send` bound that can't be met under Wasm. That's why we forked this module.

Reply:
> Did you try asking them

If by "them" you mean `integer-encoding` maintainers, then: no! We didn't try. We should. Someone should do that. It would be the right thing to do....

:sweat_smile:

Reply:
> Ahh, yah, I think the `integer-encoding` crate [enforces a `Send` bound](https://github.com/dermesser/integer-encoding-rs/blob/0250d522dcdadd4f5390838c1e31bf61782547e6/src/reader.rs#L69) that can't be met under Wasm. That's why we forked this module.

Yeah that's likely because of `async_trait`s `Send` requirements by default. But using https://docs.rs/async-trait/latest/async_trait/#non-threadsafe-futures it should be possible to avoid that.

Reply:
Created https://github.com/dermesser/integer-encoding-rs/pull/32 which fixes this issue

Reply:
also updated this PR to use that patch

Reply:
does a git branch ref prevent us from publishing an updated version of this crate?

Reply:
> does a git branch ref prevent us from publishing an updated version of this crate?

It does. It is currently unclear if `integer-encoding` is still maintained unfortunately. @flub you reviewed various varint implementations, any suggestions to what we could switch?

Reply:
I mostly compared to a multiformat's varint with a dead simple re-implementation of QUIC's varint format (which was faster).  Sadly I don't think any crate tidily exports QUIC's varint but it's super simple to write by hand: https://github.com/n0-computer/varint-bench/blob/main/src/lib.rs

Reply:
Humble suggestion: you can file an issue to address this in the future and commit the change as-is. In the mean time if there are any problems with it, you can always throw your hands up and rightly claim it was my choice to partially fork `integer-encoding`.

Alternatively: you can publish your patched crate and use that.

Reply:
aaalright, found a much cleaner version, we now use `unsigned-varint` which is the same dependency that `multihash` and friends already use.